libs/ledger-live-common/src/families/cardano/logic.ts

```diff
@@ -0,0 +1,442 @@
+    const stakingKeyPath = (
+      address.stakeCredential as TyphonTypes.HashCredential
+    ).bipPath as TyphonTypes.BipPath;
+
+    const paymentKeyPathString = getBipPathString({
+      account: accountIndex,
+      chain: paymentKeyPath.chain,
+      index: paymentKeyPath.index,
+    });
+    const stakingKeyPathString = getBipPathString({
+      account: accountIndex,
+      chain: stakingKeyPath.chain,
+      index: stakingKeyPath.index,
+    });
+
+    destination = {
+      type: TxOutputDestinationType.DEVICE_OWNED,
+      params: {
+        type: AddressType.BASE_PAYMENT_KEY_STAKE_KEY,
+        params: {
+          spendingPath: str_to_path(paymentKeyPathString),
+          stakingPath: str_to_path(stakingKeyPathString),
+        },
+      },
+    };
+  } else {
+    const address = output.address;
+    destination = {
+      type: TxOutputDestinationType.THIRD_PARTY,
+      params: {
+        addressHex: address.getHex(),
+      },
+    };
+  }
+
+  const tokenBundle: Array<AssetGroup> = Object.values(
+    groupBy(output.tokens, ({ policyId }) => policyId)
+  ).map((tokens) => ({
+    policyIdHex: tokens[0].policyId,
+    tokens: tokens.map((token) => ({
+      assetNameHex: token.assetName,
+      amount: token.amount.toString(),
+    })),
+  }));
+
+  return {
+    amount: output.amount.toString(),
+    destination,
+    tokenBundle,
+  };
+}
+
+/**
+ * returns the formatted transactionInput for ledger cardano app
+ *
+ * @param {TyphonTypes.Input} input
+ * @param {number} accountIndex
+ * @returns {TxInput}
+ */
+export function prepareLedgerInput(
+  input: TyphonTypes.Input,
+  accountIndex: number
+): TxInput {
+  const paymentKeyPath =
+    input.address.paymentCredential.type === TyphonTypes.HashType.ADDRESS
+      ? input.address.paymentCredential.bipPath
+      : undefined;
+  return {
+    txHashHex: input.txId,
+    outputIndex: input.index,
+    path: paymentKeyPath
+      ? str_to_path(
+          getBipPathString({
+            account: accountIndex,
+            chain: paymentKeyPath.chain,
+            index: paymentKeyPath.index,
+          })
+        )
+      : null,
+  };
+}
+
+export function getAccountStakeCredential(
+  xpub: string,
+  index: number
+): StakeCredential {
+  const accountXPubKey = getExtendedPublicKeyFromHex(xpub);
+  const keyPath = getCredentialKey(
+    accountXPubKey,
+    getBipPath({
+      account: index,
+      chain: StakeChain.stake,
+      index: STAKING_ADDRESS_INDEX,
+    })
+  );
+  return {
+    key: keyPath.key,
+    path: keyPath.path,
+  };
+}
+
+export function getOperationType({
+  valueChange,
+  fees,
+}: {
+  valueChange: BigNumber;
+  fees: BigNumber;
+}): OperationType {
+  return valueChange.isNegative()
+    ? valueChange.absoluteValue().eq(fees)
+      ? "FEES"
+      : "OUT"
+    : valueChange.isPositive()
+    ? "IN"
+    : "NONE";
+}
+
+export function isTestnet(currency: CryptoCurrency): boolean {
+  return currency.id === CARDANO_TESTNET_CURRENCY_ID;
+}
```

Comment:
We have a mechanism available to do that, you should replace calls to this function with:
`getCryptoCurrencyById(currencyId).isTestnetFor`

Then you can remove this function and `CARDANO_TESTNET_CURRENCY_ID`